tools/sancov/sancov.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import "fmt"
+import "flag"
+import "os"
+import "encoding/binary"
+
+const (
+	magic32SecondHalf uint32 = 0xFFFFFF32
+	magic64SecondHalf uint32 = 0xFFFFFF64
+	magicFirstHalf    uint32 = 0xC0BFFFFF
+)
+
+func usage() {
+	progName := os.Args[0]
+	fmt.Printf("Usage:\n")
+	fmt.Printf(" %s print FILE [FILE...]\n", progName)
+	os.Exit(-1)
+}
+
+func readMagicAndReturnBitness(f *os.File) uint8 {
+	header := [...]uint32{0, 0}
+	if err := binary.Read(f, binary.LittleEndian, &header); err != nil {
+		panic(err)
+	}
+
+	if header[1] != magicFirstHalf {
+		panic(fmt.Errorf("File %s has corrupted header: %x", f.Name(), header))
+	}
+
+	switch header[0] {
+	default:
+		panic(fmt.Errorf("File %s has corrupted header: %x", f.Name(), header))
+	case magic32SecondHalf:
+		return 32
+	case magic64SecondHalf:
+		return 64
+	}
+}
+
+func readOneFile(path string) []uint64 {
+	f, err := os.Open(path)
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			panic(err)
```

Comment:
We can actually gracefully recover from this particular error by not crashing. We did not write to the file, so Close errors are not fatal.